[fix/pdf-editing] Fix PDF editing

felix-schwarz opened the pull request:
## Description
- fixes bug that prevents changes to PDFs being saved in place.
- after editing, the viewer no longer asks if the PDF file should be refreshed, but instead the viewer refreshes it automatically to reflect the changes.

## Related Issue
https://github.com/owncloud/enterprise/issues/4934

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

felix-schwarz committed:
- update SDK with fix for enterprise#4934.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/owncloud/ios-app?pullRequest=1083) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/owncloud/ios-app?pullRequest=1083) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/owncloud/ios-app?pullRequest=1083) it.</sub>

felix-schwarz committed:
- DisplayViewController: add .allowUpdatesUntilLocalModificationPersists to temporarily allow updates to the viewed file until the rendered item again is stored remotely and no longer a local modification
- DocumentEditingAction: when invoked from a PDFViewerViewController, pass it along to EditDocumentViewController, so it can make use of allowUpdatesUntilLocalModificationPersists
- EditDocumentViewController: set allowUpdatesUntilLocalModificationPersists of a hosting PDFViewerViewController to true when a file is edit in place and changes overwrite the original

jesmrec commented:
About this one:

- The original problem reported in [central](https://central.owncloud.org/t/cant-save-edited-pdfs-in-owncloud-error-35/35684) is not reproducible on my side. I can edit and save overwritten files correctly with `11.8.1`. Tested with two different devices (iPad/iPhone) and different version. Also i tried different combinations of operations (overwrite + save, save + overwrite, save + overwrite + save etc...). I got never the error 35.

- The dialog `Showing latest version` only appears now if the file is updated externally, not after a local update  ✅  ✅ 

So, we can go with the approach here to get extra feedback from the reporter.
